spec/models/program_session_spec.rb

```diff
@@ -1,5 +1,145 @@
 require 'rails_helper'
 
-RSpec.describe ProgramSession, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+describe ProgramSession do
+
+  let(:proposal) { create(:proposal_with_track, :with_two_speakers) }
+
+  it "responds to .create_from_proposal" do
+    expect(ProgramSession).to respond_to(:create_from_proposal)
+  end
+
+  describe ".create_from_proposal" do
+
+    it "program session and proposal share the same title" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.title).to eq(proposal.title)
+    end
+
+    it "program session and proposal share the same abstract" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.abstract).to eq(proposal.abstract)
+    end
+
+    it "program session and proposal share the same event" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.event_id).to eq(proposal.event_id)
+    end
+
+    it "program session and proposal share the same session format" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.session_format_id).to eq(proposal.session_format_id)
+    end
+
+    it "program session and proposal share the same track" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.track_id).to eq(proposal.track_id)
+    end
+
+    it "program session has proposal id" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.proposal_id).to eq(proposal.id)
+    end
+
+    it "program session is active" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.state).to eq("active")
+    end
+
+    it "sets program session id for all speakers" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      proposal.speakers.each do |speaker|
```

Comment:
Couple problems here: you're testing the speaker values for these speakers that are already loaded. One way to check whether the changes that were made were successfully saved to the database would be to load the speakers collection from the session object. Since the has_many :speakers association probably doesn't exist on this branch yet (I think we added that to #135?), you'll have to add that to ProgramSession first.